pkg/operator/configobserver/oauth/observe_tokenconfig.go

```
@@ -0,0 +1,191 @@
+}
+
+// ObserveAccessTokenInactivityTimeoutSeconds returns the observed config for access token inactivity timeout.
+// If the function fails to read both AccessTokenMaxAgeSeconds from existing config and OAuth cluster config, it return empty config.
+// If the function only fails to read OAuth cluster config, it returns existing config.
+// If token inactivity timeout is not set in OAuth cluster config, it return empty config.
+// In any other case, it uses the value present in OAuth cluster config.
+func ObserveAccessTokenInactivityTimeoutSeconds(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	errs = []error{}
+	listers := genericlisters.(OAuthLister)
+	observedConfig := map[string]interface{}{}
+	accessTokenInactivityTimeoutPath := append(tokenConfigPath, fieldAccessTokenInactivityTimeoutSeconds)
+	defer func() {
+		// Prune the observed config so that it only contains access token inactivity timeout field.
+		ret = configobserver.Pruned(ret, accessTokenInactivityTimeoutPath)
+	}()
+
+	existingAccessTokenInactivityTimeoutSeconds, _, err := unstructured.NestedFieldCopy(existingConfig, accessTokenInactivityTimeoutPath...)
+	if err != nil {
+		// If failed to parse existing config for access token inactivity timeout, continue execution with the
+		// hope to find the value for it in OAuth cluster config.
+		errs = append(errs, err)
+	}
+
+	var existingAccessTokenInactivityTimeoutSecondsFloat *float64
+	if existingAccessTokenInactivityTimeoutSeconds != nil {
+		if f, ok := existingAccessTokenInactivityTimeoutSeconds.(float64); ok {
+			existingAccessTokenInactivityTimeoutSecondsFloat = &f
+		}
+	}
+
+	oauthConfig, err := listers.OAuthLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warning("oauth.config.openshift.io/cluster: not found")
+		return buildUnstructuredTokenConfig("", nil), nil
```

Why not return the error slice?

---

Done.